jetty-server/src/main/java/org/eclipse/jetty/server/SecureRequestCustomizer.java

```diff
@@ -244,7 +245,9 @@ protected void customize(SSLEngine sslEngine, Request request)
         if (isSniRequired() || isSniHostCheck())
         {
             String sniHost = (String)sslSession.getValue(SslContextFactory.Server.SNI_HOST);
-            X509 cert = new X509(null, (X509Certificate)sslSession.getLocalCertificates()[0]);
+            Certificate[] certificates = sslSession.getLocalCertificates();
```

Comment:
Isn't much simpler to throw early here, rather than doing `:?` and null checks later?

Reply:
yep.  Also I think the X509 instance should be cached, as it does a moderate amount of work in its constructor!